tests/src/application/validation/validators/name-validation.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { InvalidParamError } from '@/application/errors'
+import { NameValidation, type NameValidator } from '@/application/validation'
+
+interface SutTypes {
+  sut: NameValidation
+  nameValidatorStub: NameValidator
+}
+
+const makeNameValidatorStub = (): NameValidator => {
+  class NameValidatorStub implements NameValidator {
+    isValid (name: string): boolean {
+      return true
+    }
+  }
+  return new NameValidatorStub()
+}
+
+const makeSut = (): SutTypes => {
+  const fakeFieldName: string = 'fieldName'
+  const nameValidatorStub = makeNameValidatorStub()
+  const sut = new NameValidation(fakeFieldName, nameValidatorStub)
+  return {
+    sut,
+    nameValidatorStub
+  }
+}
+
+describe('NameValidation', () => {
+  it('should returns invalidParamError if fieldName is not a string', () => {
+    const { sut } = makeSut()
+
+    const result = sut.validate({ fieldName: 11 })
```

Comment:
Same in this file.
https://github.com/PetJournal/petjournal.api/pull/40#discussion_r1324466939